Content.Server/Interaction/InteractionPopupSystem.cs

```diff
@@ -0,0 +1,86 @@
+using Content.Server.Popups;
+using Content.Server.Interaction.Components;
+using Content.Shared.Audio;
+using Content.Shared.Interaction;
+using Content.Shared.MobState;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Player;
+using Robust.Shared.Timing;
+using Robust.Shared.Random;
+
+
+namespace Content.Server.Interaction;
+
+public sealed class InteractionPopupSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<InteractionPopupComponent, InteractHandEvent>(OnInteractHand);
+        SubscribeLocalEvent<InteractionPopupComponent, MobStateChangedEvent>(OnMobStateChanged);
+    }
+
+    private void OnMobStateChanged(EntityUid uid, InteractionPopupComponent component, MobStateChangedEvent args)
+    {
+        var originalSuccessChance = component.SuccessChance; // Stores this for later in case mob state changes back and forth.
+
+        if (!args.Component.IsAlive()) // if the mob being interacted with is not alive (dead, incapacitated, etc.)
+        {
+            component.SuccessChance = -1.0f; // set to the "invalid" value, which suppresses all popups and sound effects.
+            return;
+        }
```

Comment:
also skeptical that it should be done this way at all, rather than just getting `MobStateComponent` and checking manually every time someone tries to interact, that way no bugs of this nature can arise

Reply:
Very good suggestion, implementing it now.

Reply:
I got it working, but had to Old Yeller a corgi a bunch of times in testing 😭
Seems to have the intended behaviour (no sound effect or popup when animal is dead or in crit)